### PR TITLE
[GLUTEN-4594][SUB-3] Refactor Gluten Spark34 Unit Test

### PR DESCRIPTION
**Pull request by zwangsheng**
## What changes were proposed in this pull request?

Refactor Gluten Spark34 Unit Test with unify apis.

## How was this patch tested?

Exists Unit Tests

Last Commit:

```
2024-02-22T04:38:17.6615501Z [36mRun completed in 50 minutes, 10 seconds.[0m
2024-02-22T04:38:17.6617494Z [36mTotal number of tests run: 6537[0m
2024-02-22T04:38:17.6619231Z [36mSuites: completed 240, aborted 0[0m
2024-02-22T04:38:17.6621682Z [36mTests: succeeded 6537, failed 0, canceled 2, ignored 1048, pending 0[0m
2024-02-22T04:38:17.6623955Z [32mAll tests passed.[0m
```

### CODE REVIEW

File: gluten-ut/spark34/src/test/scala/org/apache/spark/sql/GlutenCachedTableSuite.scala [hunk 1]
```diff
@@ -33,7 +33,7 @@ class GlutenCachedTableSuite
     super.sparkConf.set(GlutenConfig.COLUMNAR_TABLE_CACHE_ENABLED.key, "true")
   }
 
-  test("GLUTEN - InMemoryRelation statistics") {
+  testGluten("InMemoryRelation statistics") {
```
Comment (zhztheplayer) [inline gluten-ut/spark34/src/test/scala/org/apache/spark/sql/GlutenCachedTableSuite.scala:36]: @zwangsheng Just curious that, are you using IDEs? It occurs to be that Intellij Idea's scala tester doesn't resolve `testGluten` cases?

Comment (zwangsheng) [reply gluten-ut/spark34/src/test/scala/org/apache/spark/sql/GlutenCachedTableSuite.scala:36]: Yes, it couldn't be triggered by the button, but maven tests were fine

**Comment by zwangsheng** [comment]
Here is the PR unit test for Spark34
```
2024-02-23T04:19:29.7246464Z [36mRun completed in 50 minutes, 30 seconds.[0m
2024-02-23T04:19:29.7248452Z [36mTotal number of tests run: 6538[0m
2024-02-23T04:19:29.7250213Z [36mSuites: completed 240, aborted 0[0m
2024-02-23T04:19:29.7252637Z [36mTests: succeeded 6538, failed 0, canceled 2, ignored 1048, pending 0[0m
```

**Comment by zwangsheng** [comment]
Hi @zhztheplayer @ulysses-you @PHILO-HE PLZ take a review, if there are any  concerns, please let me know.

**Review by PHILO-HE — APPROVED** [review]
Looks good! Thanks for your continuous efforts!

**Review by zhztheplayer — APPROVED** [review]
Thanks!

**Review by ulysses-you — APPROVED** [review]

**Pull request merged**